File: 77. 组合/Solution.py
class Solution:
    def backtracking(self, index, n, k, path, ans):
        if len(path) == k:
            ans.append(path[:]) # 这里需要复制path的内容，不然会加入path的引用
            return
        for i in range(index, n + 1):
            path.append(i)
            self.backtracking(i+1, n, k, path, ans)
            path.pop()

    def combine(self, n: int, k: int) -> List[List[int]]:
        ans = []
        path = []
        self.backtracking(1, n, k, path, ans)
        return ans

File: 77. 组合/Solution.cpp
#include<bits/stdc++.h>

using namespace std;


class Solution {
public:
    vector<vector<int>> ans;
    vector<int> path;
    vector<vector<int>> combine(int n, int k) {
        backtracking(1, n, k);
        return ans;
    }
    void backtracking(int index, int n, int k){
        if(path.size() == k){
            ans.push_back(path);
            return;
        }
        for(int i = index; i <= n; i++){
            path.push_back(i);
            backtracking(i+1, n, k);
            path.pop_back();
        }
    }
};